notebooks_preprocess.py

```diff
@@ -0,0 +1,74 @@
+import glob
+import json
+import os
+import re
+
+
+def format_script_for_cell(path):
+    """Reads and formats a .py file to be inserted into the json for a cell."""
+    cell_content = '\n# Cell content replaced by notebook_preprocess.py\n'
+    with open(path) as f:
+        for line in f:
+            cell_content += line
+    return cell_content
+
+
+def find_load_magics_in_cell(cell):
+    """Find the load magics in a cell and return them as a list."""
+    cell_source = cell['source']
+    load_magics = []
+    for cell_source_line in cell_source:
+        m = re.match('#\s?%load.*', cell_source_line)
+        if m:
+            load_magics.append(m.group())
+    return load_magics
+
+
+def get_cell_content_as_string(cell):
+    """Return the cells source as a single string."""
+    cell_source = cell['source']
+    cell_string = ''
+    for line in cell_source:
+        cell_string += line
+    return cell_string + '\n'
+
+def process_cell(path, cell):
+    """Append the data from the load magics into the cell content."""
+    modified = False
+    # See if there are any load magics used
+    load_magics = find_load_magics_in_cell(cell)
+
+    # Replace the load magics with content from their recpective files
+    for magic_string in load_magics:
+        script_path = os.path.dirname(path) + os.sep + magic_string.split('load ')[1]
+        formatted_script = format_script_for_cell(script_path)
+        cell_str = get_cell_content_as_string(cell)
+        cell['source'] = cell_str + formatted_script
+        modified = True
+    return modified, cell
+
+
+# Recursively grab all notebooks and process them
+notebooks = glob.glob('notebooks' + os.sep + '**' + os.sep + '*.ipynb', recursive=True)
+for notebook in notebooks:
+    modified = False
+    # Read in the notebook as JSON data
+    notebook_modified = False
+    print('Reading notebook: {}'.format(notebook))
+    with open(notebook, 'r') as f:
+        json_data = json.load(f)
+
+    # Process each cell in the file
+    cells = json_data['cells']
+
+    for cell in cells:
+        cell_modified, cell = process_cell(notebook, cell)
```

Comment:
Technically looks like there's no need to return `cell`. That could simplify the loop:
```python
for cell in json_data['cells']:
    modified = process_cell(notebook, cell) or modified
```